sql: TestCancelDistSQLQuery failed

sql.TestCancelDistSQLQuery [failed](https://mesolite.cluster.engflow.com/invocations/default/e8bc8fb6-a551-4d1f-83d1-930b9f89d43c?testReportRun=3&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [6e3c4b94d8f56ce5a436d53e6a1d7e12ddc5a420](https://github.com/cockroachdb/cockroach/commits/6e3c4b94d8f56ce5a436d53e6a1d7e12ddc5a420):


```
=== RUN   TestCancelDistSQLQuery
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestCancelDistSQLQuery2772578303
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:155: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    run_control_test.go:134: pq: could not cancel query 187a552a2cd9fc2d0000000000000001: 
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestCancelDistSQLQuery2772578303
--- FAIL: TestCancelDistSQLQuery (9.29s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCancelDistSQLQuery.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-57181

--- a/pkg/sql/run_control_test.go
+++ b/pkg/sql/run_control_test.go
@@ -130,7 +130,10 @@ func TestCancelDistSQLQuery(t *testing.T) {
 		errChan <- err
 	}()
 	_, err := conn2.Exec(cancelQuery)
-	if err != nil && !testutils.IsError(err, "query ID") {
+	// We might have blocked the cancellation query longer than it took the
+	// target query to execute, in which case the cancel query itself errors out
+	// (and it's ok).
+	if err != nil && !testutils.IsError(err, "could not cancel query") {
 		t.Fatal(err)
 	}
 

